app/controllers/admin/agent_agendas_controller.rb

```diff
@@ -9,4 +9,17 @@ def show
     @selected_event_id = params[:selected_event_id]
     @date = params[:date].present? ? Date.parse(params[:date]) : nil
   end
+
+  def update
```

Comment:
Ça aurait pas plutôt sa place dans http://localhost:5000/agents/preferences? Tu y as peut-être déjà pensé?

Reply:
J'y ai pensé, et il y a deux détails qui m'ont fait pencher pour l'agent_agenda_controller : 
- ce bouton pour afficher ou cacher les samedis s'applique à tous les agendas que regarde l'agent, pas juste le sien. Sémantiquement, ça donne l'impression que c'est donc une action sur les `agent_agenda`, et pas vraiment sur ses préférences à lui. (Ce fonctionnement est pensé pour le cas d'usage où un cnfs qui ne travaille pas le samedi consulte l'agenda d'un collègue qui travaille le samedi).
- en mettant cette action dans le controller de agent_preferences, on est plus proche de ce qui se passe en base de données, mais plus loin de l'interface : cette action est assez liée au show puisqu'elle garde les paramètres de date, status et selected_event_id pour faire la redirection correctement.

Dans le doute, je viens d'ajouter quelques commits dans une autre branche pour voir à quoi ça ressemblerait si on déplaçait cette action : https://github.com/betagouv/rdv-solidarites.fr/compare/2145-samedi-dans-calendrier...2145-samedi-dans-calendrier-in-agents-preferences-controller

Je crois qu'avec l'interface telle qu'elle existe, j'ai l'impression que ça a plus de sens de garder cette action ici, mais je serais content d'en parler de vive voix aussi. :)